### PR TITLE
VACMS-11968: Hub list as menu

### PR DESCRIPTION
**Pull request by dsasser**
## Description

Relates to #11968

## Screenshots
New menu:
Screen Shot 2023-02-03 at 12 58 56 PM

Editing a Link Logged in as Admin:
screencapture-va-gov-cms-ddev-site-admin-structure-menu-item-11723-edit-2023-02-03-13_00_20

Editing a LInk Logged in as Content Admin:
Screen Shot 2023-02-03 at 1 02 39 PM

### Expected Menu Order
Health Care
Disability
Education and Training
Careers and employment
Pension
Housing assistance
Life insurance
Burials and memorials
Records
Service member benefits
Family member benefits
VA Dept Information

As user with admin role
1. Navigate to the menu at: https://cms-pfcgaquu99xi4wzksftxmvukusulz3bh.ci.cms.va.gov//admin/structure/menu/manage/home-page-hub-list
   - [x] Validate that the new menu links have been created (this is done via Drush deploy hook)
   - [x] Validate the menu links are in the order described above
2. Then edit any link
   - [x] Validate that the 'View mode' and 'Parent link' fields are visible
   - [x] Validate that the 'attributes' field is not visible.
3. Then validate Acceptance Criteria from issue:
   - [x] Menu Administrative summary: `Menu on VA.gov new home page that displays Benefit Hubs`
   - [x] New custom field: `Hub list icon` - select list that will not re-use any existing icon lists, and will include `VA Department Info`. Required field, with no "None" option.
   - [x] New custom field: `Link summary` - textarea, max 180 chars. Required.
   - [x] `Menu link title` limited to 25 characters (per SW Content team; same mural). Required.
   - [x] Helper text included per design

As user with content_admin role
1. Navigate to the menu at: https://cms-pfcgaquu99xi4wzksftxmvukusulz3bh.ci.cms.va.gov//admin/structure/menu/manage/home-page-hub-list
2. Edit any link
   - [x] Validate that the 'attributes', "View mode" and "Parent link" fields is are not visible.

### Definition of Done
- [x] Tests have been added.
- [x] Automated tests have passed.
- [x] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [x] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ module:
   content_lock_timeout: 0
   content_moderation: 0
   contextual: 0
-  core_event_dispatcher: 0
+  core_event_dispatcher: 1
```
Comment (dsasser) [inline config/sync/core.extension.yml:39]: This was needed to ensure that the forms event subscriber fired after the menu_link_attributes form alter.

**Review by ndouglas — APPROVED** [review]
Screen Shot 2023-02-07 at 12 04 12 PM
Screen Shot 2023-02-07 at 12 04 38 PM
Screen Shot 2023-02-07 at 12 04 43 PM
Screen Shot 2023-02-07 at 12 04 52 PM
Screen Shot 2023-02-07 at 12 09 37 PM
Screen Shot 2023-02-07 at 12 10 07 PM
Screen Shot 2023-02-07 at 12 16 57 PM

This all looks right to me.  Tentatively approving because I'm out of my element.

File: docroot/modules/custom/va_gov_home/va_gov_home.deploy.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    74,
+    77,
+    78,
+    79,
+    809,
+    810,
+  ];
+
+  // Create internal menu links.
+  foreach ($nids as $weight => $nid) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_home/va_gov_home.deploy.php:54]: I'm confused about the the key on the Weight using a flat array.  I am guessing the order in the arrray is the desired order so the default index is fine as the weight.  If my assumption is right, no need to do anything.  Just wanted to confirm that is desired.

**Review by swirtSJW — COMMENTED** [review]
Code wise this looks great.  I'm having trouble connecting to socks while in the car.  I will test this as soon as I get back to my desk.

**Comment by davidmpickett** [comment]
I logged in as a Content Admin and can confirm the fields did not show up
screencapture-cms-pfcgaquu99xi4wzksftxmvukusulz3bh-ci-cms-va-gov-admin-structure-menu-item-11723-edit-2023-02-07-15_13_58

File: docroot/modules/custom/va_gov_home/va_gov_home.deploy.php (hunk 2 above)
Comment (dsasser) [reply docroot/modules/custom/va_gov_home/va_gov_home.deploy.php:54]: >  I am guessing the order in the arrray is the desired order so the default index is fine as the weight.

That is correct. I was being lazy and using PHP's auto indexing to create the necessary weight, but these are in the order we want them to appear.

File: docroot/modules/custom/va_gov_home/src/EventSubscriber/FormEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    if (!empty($form['menu_parent'])) {
+      $form['menu_parent']['#access'] = $admin;
+    }
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
```
[the remark this answers is not in the document]
Comment (dsasser) [reply docroot/modules/custom/va_gov_home/src/EventSubscriber/FormEventSubscriber.php:128]: I didn't think about that but it makes sense to me.

**Review by swirtSJW — APPROVED** [review]
All looks and tests out great.  Nice work @dsasser 

image

**Pull request merged**